packages/elastic-apm-synthtrace/src/lib/apm/base_span.ts

```diff
@@ -24,7 +24,7 @@ export class BaseSpan extends Serializable<ApmFields> {
     });
   }
 
-  parent(span: BaseSpan) {
+  parent(span: BaseSpan): this {
```

Comment:
is the explicit return type annotation necessary?

Reply:
Yes, this allows `parent()` to be called anywhere in the fluent chain and will always return the most concrete instance. Otherwise chaining to `parent()` will scope the DSL down to only those methods available on `BaseSpan`.